src/gui/sdlmain.cpp

```diff
@@ -1849,24 +1858,45 @@ uint8_t GFX_SetSize(const int width, const int height,
 					}
 					glAttachShader(sdl.opengl.program_object, vertexShader);
 					glAttachShader(sdl.opengl.program_object, fragmentShader);
+
 					// Link the program
 					glLinkProgram(sdl.opengl.program_object);
+
 					// Even if we *are* successful, we may delete the shader objects
 					glDeleteShader(vertexShader);
 					glDeleteShader(fragmentShader);
 
 					// Check the link status
-					GLint isProgramLinked;
-					glGetProgramiv(sdl.opengl.program_object, GL_LINK_STATUS, &isProgramLinked);
-					if (!isProgramLinked) {
-						GLint info_len = 0;
-						glGetProgramiv(sdl.opengl.program_object, GL_INFO_LOG_LENGTH, &info_len);
-
-						if (info_len > 1) {
-							std::vector<GLchar> info_log(info_len);
-							glGetProgramInfoLog(sdl.opengl.program_object, info_len, nullptr, info_log.data());
-							LOG_ERR("SDL:OPENGL: Error link program:\n %s", info_log.data());
+					GLint is_program_linked = 0;
+					glGetProgramiv(sdl.opengl.program_object, GL_LINK_STATUS, &is_program_linked);
+
+					// The info log might contain warnings and info messages
+					// even if the linking was successful, so we'll always log
+					// it if it's non-empty.
+					GLint info_len = 0;
+
+					glGetProgramiv(sdl.opengl.program_object,
+					               GL_INFO_LOG_LENGTH,
+					               &info_len);
+
+					if (info_len > 1) {
+						std::vector<GLchar> info_log(info_len);
+
+						glGetProgramInfoLog(sdl.opengl.program_object,
+						                    info_len,
+						                    nullptr,
+						                    info_log.data());
+
+						if (is_program_linked) {
+							LOG_WARNING("SDL:OPENGL: Program info log:\n %s",
+							            info_log.data());
```

Comment:
Just a question: if the program is linked, then the presence of the  message indicates some kind of warning?

(Or on other words: there are no messages when things are perfect?)

Reply:
We don't know, it's entirely driver dependent. But usually they are either warnings or helpful hints to improve things, so their presence indicates "warning like" scenarios.

Reply:
Thanks - yup, I agree `LOG_WARNING` sounds like a good fit (because these aren't just bog-standard "sunny day" messages, if they exist, despite the program compiling).

Hopefully @Burrito78's GLSL 1.2 system has something to say at these spots!